src/platforms/esp32/components/avm_builtins/network_driver.c

```diff
@@ -335,6 +335,7 @@ static void event_handler(void *arg, esp_event_base_t event_base, int32_t event_
             case IP_EVENT_STA_GOT_IP: {
                 ip_event_got_ip_t *event = (ip_event_got_ip_t *) event_data;
                 ESP_LOGI(TAG, "IP_EVENT_STA_GOT_IP: %s", inet_ntoa(event->ip_info.ip));
+                esp_sntp_restart();
```

Comment:
Let's add a short comment explaining why we are restarting (outside the context of this PR is not obvious at all).

Reply:
very much needed, thx.